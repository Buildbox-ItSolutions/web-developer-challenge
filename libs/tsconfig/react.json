{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "React",
  "extends": "./base.json",
  "compilerOptions": {
    "jsx": "react-jsx",
    "skipLibCheck": true,
    "lib": ["ES2015", "DOM"],
    "module": "ESNext",
    "target": "es6"
  }
}
