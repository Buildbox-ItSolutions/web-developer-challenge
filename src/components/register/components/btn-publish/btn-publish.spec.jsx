import { render, screen } from '@testing-library/react';
import { BtnPublish } from '.';

describe('<BtnPublish/>', () => {
  it('should BtnPublish render', () => {
    render(<BtnPublish disabled={true} />);

    const button = screen.getByRole('button', { name: /publicar/i });
    expect(button).toBeInTheDocument();
  });

  it('should be disabled when disable is true', () => {
    render(<BtnPublish disabled={true} />);

    const button = screen.getByRole('button', { name: /pu/i });

    expect(button).toBeDisabled();
  });

  it('should match snapshot', () => {
    const { container } = render(<BtnPublish disabled={true} />);

    expect(container).toMatchSnapshot();
  });
});
