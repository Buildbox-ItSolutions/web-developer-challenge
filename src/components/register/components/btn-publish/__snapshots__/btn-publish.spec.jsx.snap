// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BtnPublish/> should match snapshot 1`] = `
<div>
  <button
    class="btn-publish"
    disabled=""
  >
    Publicar
  </button>
</div>
`;
